skfda/exploratory/stats/_functional_transformers.py

```diff
@@ -342,3 +342,167 @@ def number_up_crossings(
+
+        Returns:
+            ndarray of shape (n_dimensions, n_samples) with the values of the
+            specified moment.
+
+    Example:
+
+    We will calculate the first unconditional moment of the Canadian Weather
+    data set. In order to simplify the example, we will use only the first
+    five samples.
+    First we proceed to import the data set.
+    >>> from skfda.datasets import fetch_weather
+    >>> X = fetch_weather(return_X_y=True)[0]
+
+    Then we call the function with the samples that we want to consider and the
+    specified moment order.
+    >>> import numpy as np
+    >>> from skfda.exploratory.stats import unconditional_moments
+    >>> np.around(unconditional_moments(X[:5], 1), decimals=2)
+    array([[ 4.7 ,  4.03],
+           [ 6.16,  3.96],
+           [ 5.52,  4.01],
+           [ 6.82,  3.44],
+           [ 5.25,  3.29]])
+    """
+    return unconditional_expected_value(
+        data,
+        lambda x: np.power(x, n),
+    )
+
+
+def unconditional_expected_value(
+    data: Union[FDataBasis, FDataGrid],
+    function: Callable[[np.ndarray], np.ndarray],
+) -> NDArrayFloat:
+    r"""
+    Calculate the unconditional expected value of a function.
+
+    Next formula shows for a defined transformation :math: `g(x(t))`
+    and p observations, how the unconditional expected values are calculated:
+    .. math::
+            f_1(x(t))=\frac{1}{\left( b-a\right)}\int_a^b g
+            \left(x_1(t)\right)dt,\dots,
+            f_p(x(t))=\frac{1}{\left( b-a\right)}\int_a^b g
+            \left(x_p(t)\right) dt
+        Args:
+            data: FDataGrid or FDataBasis where we want to calculate
+            the expected value.
+            f: function that specifies how the expected value to is calculated.
+            It has to be a function of X(t).
+        Returns:
+            ndarray of shape (n_dimensions, n_samples) with the values of the
+            expectations.
+
+    Example:
+    We will use this funtion to calculate the logarithmic first moment
+    of the first 5 samples of the Berkeley Growth dataset.
+    We will start by importing it.
+    >>> from skfda.datasets import fetch_growth
+    >>> X = fetch_growth(return_X_y=True)[0]
+
+    We will define a function that calculates the inverse first moment.
+    >>> import numpy as np
+    >>> f = lambda x: np.power(np.log(x), 1)
+
+    Then we call the function with the dataset and the function.
+    >>> from skfda.exploratory.stats import unconditional_expected_value
+    >>> np.around(unconditional_expected_value(X[:5], f), decimals=2)
+        array([[ 4.96],
+               [ 4.88],
+               [ 4.85],
+               [ 4.9 ],
+               [ 4.84]])
+    """
+    domain_range = np.prod(
+        [
+            (iterval[1] - iterval[0])
+            for iterval in data.domain_range
+        ],
+    )
+
+    if isinstance(data, FDataGrid):
+        return function(data).integrate() / domain_range
+
+    return nquad_vec(
```

Comment:
I don't think this is ok. You seem to compute the integral of the function instead of the function applied to the random variable. Please add a test using the same data in both discretized and basis form to check if it works.